backends/bmv2/common/expression.cpp

```diff
@@ -298,6 +304,8 @@ void ExpressionConverter::postorder(const IR::Member* expression)  {
             result->emplace("value", fieldName);
         } else if (parentType->is<IR::Type_HeaderUnion>()) {
             auto l = get(expression->expr);
+            if (!l)
```

Comment:
Would it make sense to convert this into a macro? So that this check does not have to be implemented manually for all these calls.

Reply:
Well, you can create a macro that also declares the variable, like this:
`#define GET2(var, exp) auto var = get(exp); if (!var) return;`
But this does not make it obvious that GET2 declares a new identifier when you invoke it:
`GET2(l, expression);`. It will also generate 3 statements, not just one. You cannot use the do-while trick in macros to generate a single statement, since you need `var` to be declared in the same scope.

The alternative solution is to use a GCC extension, which uses statement expressions:
```
#define GET1(expression) ({ \
    auto r = get(expression); \
    if (!r) return; \
    r; \
})
```
However, this may give wrong results if the `r` variable is shadowed.
So I don't like any of these solutions using a macro. Do you have an alternative proposal?

Reply:
No unfortunately not. There does not seem a nice way to implement this. But since the changes are localized to a single file the impact seems low.